src/subdevice/configuration.rs

```diff
@@ -335,20 +287,12 @@ where
         let start_offset = *global_offset;
         // let mut total_bit_len = 0;
 
-        for (sync_manager_index, (sm_type, sync_manager)) in self
-            .state
-            .config
-            .mailbox
-            .coe_sync_manager_types
-            .iter()
-            .zip(sync_managers.iter())
-            .enumerate()
-        {
+        for (sync_manager_index, sync_manager) in sync_managers.iter().enumerate() {
             let sync_manager_index = sync_manager_index as u8;
 
             let sm_address = SM_BASE_ADDRESS + u16::from(sync_manager_index);
 
-            if *sm_type != desired_sm_type {
+            if sync_manager.usage_type != desired_sm_type {
```

Comment:
There's a method that should be used instead of the field. Crappy code on my part, sorry!

```suggestion
            if sync_manager.usage_type() != desired_sm_type {
```